src/main/java/de/retest/recheck/ui/descriptors/RootElementFilter.java

```diff
@@ -0,0 +1,93 @@
+package de.retest.recheck.ui.descriptors;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import de.retest.recheck.ignore.Filter;
+import de.retest.recheck.ui.image.Screenshot;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class RootElementFilter {
+	private final Filter filter;
+
+	public SutState filter( final SutState goldenMaster ) {
```

Comment:
A SUT state is not necessarily a golden master, whereas a golden master is always a SUT state. Consider expected and actual. Both are SUT state, but only expected has been approved by a human, so only this is the golden master.

```suggestion
	public SutState filter( final SutState sutState ) {
```